Query/EncloseNames.php

```diff
@@ -0,0 +1,142 @@
+ * ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDERS AND CONTRIBUTORS BE
+ * LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+ * CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+ * SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+ * INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+ * CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ * POSSIBILITY OF SUCH DAMAGE.
+ */
+
+namespace Hoa\Database\Query;
+
+/**
+ * Trait \Hoa\Database\Query\EncloseNames.
+ *
+ * Enclose names feature.
+ *
+ * @copyright  Copyright © 2007-2015 Hoa community
+ * @license    New BSD License
+ */
+trait EncloseNames
+{
+    /**
+     * State of the enclosing: Either true for enable the enclosing feature
+     * or false for disable it.
+     *
+     * @var bool
+     */
+    protected $_enableEnclose = false;
+
+    /**
+     * Enclose opening symbol.
+     *
+     * @var string
+     */
+    protected $_openingSymbol = '"';
+
+    /**
+     * Enclose closing symbol.
+     *
+     * @var string
+     */
+    protected $_closingSymbol = '"';
+
+
+
+    /**
+     * Set enclose symbols.
+     *
+     * @param   string  $openingSymbol    Opening symbol.
+     * @param   string  $closingSymbol    Closing symbol.
+     * @return  \Hoa\Database\Query\EncloseNames
+     */
+    public function setEncloseSymbol($openingSymbol, $closingSymbol = null)
+    {
+        $this->_openingSymbol = $openingSymbol;
+        $this->_closingSymbol =
+            (null === $closingSymbol)
+                ? $openingSymbol
+                : $closingSymbol;
+
+        return $this;
+    }
+
+    /**
+     * Enable or disable enclosing names.
+     *
+     * @param   bool  $enable    Enable or disable.
+     * @return  bool
+     */
+    public function enableEncloseNames($enable = true)
+    {
+        $old                  = $this->_enableEnclose;
+        $this->_enableEnclose = $enable;
+
+        return $old;
+    }
+
+    /**
+     * Enclose table/column/alias names with defined symbol.
+     *
+     * @param   array|string  $names    Table/column/alias names.
+     * @return  array|string
+     */
+    protected function enclose($names)
+    {
+        if (false === $this->_enableEnclose) {
+            return $names;
+        }
+
+        if (false === is_array($names)) {
+            return $this->_enclose($names);
+        }
+
+        foreach ($names as &$name) {
+            $name = $this->_enclose($name);
+        }
+
+        return $names;
+    }
+
+    /**
+     * Enclose table/column/alias name with defined symbol.
+     *
+     * @param   string  $name    Table/column/alias name.
+     * @return  string
+     */
+    protected function _enclose($name)
+    {
+        if (0 === preg_match('#\s|\(#', $name)) {
```

Comment:
Nop, we enclose the symbol only if there is NO space (alias) or parentheses (function).